plugins/balloontoolbar/plugin.js

```diff
@@ -675,19 +675,32 @@
 				if ( this.rect.visible ) {
 					return;
 				}
-				var editable = this.editor.editable();
+				var editor = this.editor,
+					editable = editor.editable(),
+					editorScrollableElement = editable.isInline() ? editable : editable.getDocument(),
+					win = CKEDITOR.document.getWindow();
+
+				// iOS classic editor listens on frame parent element for editor `scroll` event (#1910).
+				// Since iOS 13, this `if` won't be necesary any longer https://bugs.webkit.org/show_bug.cgi?id=149264.
+				if ( CKEDITOR.env.iOS && !editable.isInline() ) {
+					editorScrollableElement = editor.window.getFrame().getParent();
+				}
+
 				this._detachListeners();
 
-				function attachListener() {
-					this.reposition();
-				}
+				this._listeners.push( editor.on( 'change', attachListener, this ) );
+				this._listeners.push( editor.on( 'resize', attachListener, this ) );
 
-				this._listeners.push( this.editor.on( 'change', attachListener, this ) );
-				this._listeners.push( this.editor.on( 'resize', attachListener, this ) );
-				this._listeners.push( CKEDITOR.document.getWindow().on( 'resize', attachListener, this ) );
-				this._listeners.push( editable.attachListener( editable.getDocument(), 'scroll', attachListener, this ) );
+				this._listeners.push( win.on( 'resize', attachListener, this ) );
+				this._listeners.push( win.on( 'scroll', attachListener, this ) );
+
+				this._listeners.push( editorScrollableElement.on( 'scroll', attachListener, this ) );
 
 				CKEDITOR.ui.balloonPanel.prototype.show.call( this );
+
+				function attachListener() {
+					this.reposition();
+				}
```

Comment:
Could we skip this middleman function and pass `this.reposition` directly? E.g.

```javascript
this._listeners.push( editor.on( 'change', this.reposition, this ) );
```

`this` object should be correctly resolved here, wdyt?

Reply:
I had to revert this change. `this.reposition` is defined in a prototype. So when there are 2 editors on a page, there might appear a situation that both of them will call `this.detachListener`. This will result with attempt to remove twice the same listener. To avoid it, it's necessary to preserve the context for a given listener, to avoid the situation where a different instance of balloon toolbar removes such listener.